save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt

```diff
@@ -97,9 +106,12 @@ internal class OrganizationController(
      */
     @PostMapping("/{organizationName}/update")
     @PreAuthorize("isAuthenticated()")
+    @Suppress("UnsafeCallOnNullableType")
     fun updateOrganization(@RequestBody organization: Organization, authentication: Authentication): Mono<StringResponse> {
         val userId = (authentication.details as AuthenticationDetails).id
-        val role = lnkUserOrganizationService.findRoleByUserIdAndOrganizationName(userId, organization.name)
+        val organizationRole = lnkUserOrganizationService.findRoleByUserIdAndOrganizationName(userId, organization.name)
+        val globalRole = userDetailsService.getGlobalRole(authentication)
+        val role = listOf(organizationRole, globalRole).maxByOrNull { it.priority }!!
```

Comment:
There is a lot of repetitions of this construction: get role, get glolbal role, keep highest. It should be extracted somewhere.

Also, it starts to concern me now: if we allow any action to superadmin, performing all the checks under the hood, it can become really easy do damage data from frontend by mistake, e.g. change change another organization's description thinking it's yours. I think we should add additional indication on frontend that such actions are performed using superadmin permissions rather than usual permissions.

Reply:
ok, will add some logic to get rid of repetitions.

Talking about superadmins doing something by mistake, I should say that we have some kind of issue for that (#706). I will implement it there because this PR is sort of bugfix. Your idea is good, but do we need to implement it taking into consideration that there will be only 6 of us who is able to damage something having superadmin roles

Reply:
> a function somewhere that gets global role and either organizationRole or projectRole

Yes, I think we can do that in backend, e.g. create a methods like `OrganizationService#getGlobalRoleOrOrganizationRole`

Regarding the second topic, ok, let's properly implement indication uner #706.

Reply:
Ok, mentioned it there.